basil/utils/USBBinds.py

```diff
@@ -31,12 +31,16 @@ def query_identification(rm, resource, baud_rate, read_termination=None, write_t
     inst.read_termination = read_termination
     inst.write_termination = write_termination
     try:
-        reply = inst.query("*IDN?", delay=0.1)
+        reply = inst.query(indentifer_cmd, delay=0.1)
     except pyvisa.VisaIOError:
         # This retries the query a second time, since some devices do not answer the first time.
         # If a second exception arrises, it will be handled in calling function.
-        reply = inst.query("*IDN?", delay=0.1)
-    return reply
+        reply = inst.query(indentifer_cmd, delay=0.1)
+        
+    if read_after_query:
+        return inst.read()
+    else:
+        return reply
```

Comment:
Why is this needed?

Reply:
Because the SHQ 122m returns as query response the command string as check if the command was executed successfully. To get the return value of an command (or in this case the identification string) one has to additionally perform a read operation.